python/zenoss/protocols/services/zep.py

```diff
@@ -29,9 +31,14 @@
 class ZepServiceException(Exception):
     pass
 
-class ZepConnectionError(ServiceConnectionError):
+class ZepConnectionTimeout(ServiceConnectionError):
     pass
 
+class ZepConnectionError(ServiceConnectionError):
+    def __init__(self, msg, e):
+        if isinstance(e, (TimeoutError, socket_timeout)):
+            raise ZepConnectionTimeout(msg, e)
```

Comment:
ow. this hurts.  Using an initializer of one class to convert the exception if it's some type rather than modifying the originally throwing code to raise one or the other.